### DISCUSSION TITLE
--root-path "/" appears to break path resolution from 0.26.0 onwards

### DISCUSSION THREAD

**Discussion by alexdunnjpl** [Kludex/uvicorn#2244] [Potential Issue]
Up until and including 0.25.0, setting a root path of `/` was valid

`uvicorn api.app:app --root-path "/" --host 0.0.0.0 --reload`

Since 0.26.0, this results in unintentional appending of `/` to all paths, resulting in paths failing to resolve.

[See further details in related FastAPI issue](https://github.com/tiangolo/fastapi/discussions/11102)

**Comment by msmitherdc** [comment]
I am having a similar issue where I am setting a root_path with django. Specifically the change from `self.scope["path"] = path` to `self.scope["path"] = full_path` in protocols/http/httptools_impl.py causes the root_path to be duplicated in my use in django. 

we use
```
          uvicorn.run(
               'asgi.GRiDasgi:application',
               host='0.0.0.0',  
               root_path=settings.URL_ROOT, 
               port=8000,
               fd=infd, 
               log_config=logconfig,
               proxy_headers=True,
               limit_max_requests=500,
          )
````
infd is set from circusd, and this is run inside an nginx proxypath that matches the root_path. This change causes the root path to be duplicated
